docs/glossary.rst

```diff
@@ -304,8 +304,8 @@ Glossary
 
    identity
      An identity is an object identifying the user associated with the
-     current request.  The identity can be any object, but should implement a
-     ``__str__`` method that outputs a corresponding :term:`userid`.
+     current request.  The identity can be any object, but security policies
+     should ensure that it represents a valid user (not deleted or deactivated).
```

Comment:
I don't think it's required here that they aren't deleted or deactivated etc but `permits` and anything else that uses it does need to be aware of whatever this opaque object represents.

Reply:
`a valid user with respect to the application (e.g. not deleted).`

Reply:
I guess my django experience showed!  The auth module there changed somewhat recently to refuse login for deactivated accounts.  I suppose it’s a convenience, but does not cover all needs: new account with unvalidated email, suspended but can still see some pages, etc.

I agree that Pyramid auth is more flexible and identify’s scope more limited: return an object or none, let other methods decide permissions.